app/src/main/java/com/univoice/feature/quickscan/QuickScanActivity.kt

```diff
@@ -0,0 +1,110 @@
+
+import android.content.Intent
+import android.view.View
+import android.view.ViewGroup
+import androidx.activity.viewModels
+import androidx.viewpager2.widget.ViewPager2
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator
+import com.univoice.R
+import com.univoice.core_ui.base.BindingActivity
+import com.univoice.databinding.ActivityQuickScanBinding
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class QuickScanActivity : BindingActivity<ActivityQuickScanBinding>(R.layout.activity_quick_scan) {
+    private val viewModel by viewModels<QuickScanViewModel>()
+    override fun initView() {
+        initToolbarClickListener()
+        initAdapter()
+        addMarginsToTabs(binding.tabQuickScan)
+        unableTabMotion()
+    }
+
+    private fun unableTabMotion() {
+        binding.tabQuickScan.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
+            override fun onTabSelected(tab: TabLayout.Tab?) {}
+            override fun onTabUnselected(tab: TabLayout.Tab) {}
+            override fun onTabReselected(tab: TabLayout.Tab) {}
+
+        })
+    }
+
+    private fun addMarginsToTabs(tabLayout: TabLayout) {
+        tabLayout.addOnLayoutChangeListener { _, _, _, _, _, _, _, _, _ ->
+            for (i in 0 until tabLayout.tabCount - 1) {
+                val tab = (tabLayout.getChildAt(0) as ViewGroup).getChildAt(i)
+                val params = tab.layoutParams as ViewGroup.MarginLayoutParams
+                when (tabLayout.tabCount) {
+                    in 2..5 -> params.setMargins(0, 0, 8, 0)
+                    in 6..8 -> params.setMargins(0, 0, 6, 0)
+                    else -> params.setMargins(0, 0, 4, 0)
+                }
+                tab.requestLayout()
+            }
+        }
+
+    }
+
+    private fun initAdapter() {
+        QuickScanAdapter() { id, isBookmark ->
+            viewModel.updateBookmark(id, isBookmark)
+        }.also {
+            binding.vpQuickScan.adapter = it
+            it.submitList(viewModel.mockQuickScanList)
+            initTabLayout()
+            initPageChangeCallback(it)
+        }
+    }
+
+    private fun initPageChangeCallback(adapter: QuickScanAdapter) {
+        var currentPos = 0
+        var currentState = ViewPager2.SCROLL_STATE_IDLE
+        binding.vpQuickScan.registerOnPageChangeCallback(object :
+            ViewPager2.OnPageChangeCallback() {
+            override fun onPageScrolled(
+                position: Int,
+                positionOffset: Float,
+                positionOffsetPixels: Int
+            ) {
+                if (currentState == ViewPager2.SCROLL_STATE_DRAGGING) {
+                    if (currentPos == adapter.itemCount - 1 && position == adapter.itemCount - 1) {
+                        val intent =
+                            Intent(this@QuickScanActivity, QuickScanCompleteActivity::class.java)
+                        startActivity(intent)
```

Comment:
```suggestion
Intent(this, SignUpActivity::class.java).apply {
    signUpLauncher.launch(this)
}
```

P2: 요런식으로 Scope를 사용하면 불필요한 변수 생성을 막을 수 있습니다!

Reply:
우왕 신기하네요